stdlib/public/SwiftShims/SwiftStddef.h

```diff
@@ -37,9 +37,18 @@ typedef __typeof__(_Generic((__swift_size_t)0,                                 \
                             unsigned short : (short)0,                         \
                             unsigned char : (signed char)0)) __swift_ssize_t;
 #elif defined(__cplusplus)
+#if __has_include (<type_traits>)
```

Comment:
This likely isn't correct on Darwin because SwiftShims aren't generally allowed to import other headers (see the comment about \<stddef.h> above). At the same time `_Generic` isn't part of C++ officially. Maybe you can just add `|| __swift__` to the `#if` above instead?

Reply:
It appears that clang allows _Generic as a c11 extension in c++ code. This seems fine then.